vnext/local-cli/config/dependencyConfig.js

```diff
@@ -1,102 +1,167 @@
+        }
+      });
+
+      const projectFile = path.join(sourceDir, projects[i].projectFile);
+
+      const projectXml = configUtils.readProjectFile(projectFile);
+
+      // Calculating (auto) items
+
+      projects[i].projectName = configUtils.getProjectName(projectXml);
+      projects[i].projectLang = configUtils.getProjectLanguage(projectFile);
+      projects[i].projectGuid = configUtils.getProjectGuid(projectXml);
+
+      if (projects[i].directDependency) {
+        // Verifying (req) items
+        requiredForDirectDepenencies.forEach(item => {
+          if (!(item in projects[i])) {
+            throw `${item} is required for each project in react-native.config`;
+          }
+        });
       }
     }
-  }
+  } else {
+    // No react-native.config, try to heurestically find the correct project
 
-  return {
-    sourceDir,
-    packageIDL,
-    packageName,
-    cppProjFile,
-    csProjectFile,
-    projGUID,
-  };
-}
+    const projectFile = configUtils.findProjectFile(sourceDir, packageJson);
 
-function findWindowsAppFolder(folder) {
-  const winDir = 'windows';
-  const joinedDir = path.join(folder, winDir);
-  if (fs.existsSync(joinedDir)) {
-    return joinedDir;
-  }
+    const projectLang = configUtils.getProjectLanguage(projectFile);
 
-  return null;
-}
+    const projectXml = configUtils.readProjectFile(projectFile);
 
-// assumption is every cpp native module will have a ReactPackageProvider.idl defined
-function findPackageProviderIDL(folder) {
-  const PackageIDLPath = glob.sync(path.join('**', 'ReactPackageProvider.idl'), {
-    cwd: folder,
-    ignore: ['node_modules/**', '**/Debug/**', '**/Release/**', 'Generated Files'],
-  })[0];
+    const projectName = configUtils.getProjectName(projectXml);
 
-  return PackageIDLPath ? path.join(folder, PackageIDLPath) : null;
-}
+    const projectGuid = configUtils.getProjectGuid(projectXml);
 
-// look for packagename 'XYZ' in string 'namesapce XYZ {'
-function parsePackageIDLFile(packageIDL) {
-  const buf = fs.readFileSync(packageIDL, 'utf8');
-  const indexofNameSpace = buf.indexOf('namespace') + 9;
-  const indexofBracket = buf.indexOf('{');
-  const packageName = buf.substring(indexofNameSpace, indexofBracket).replace(/\s+/g, ' ').trim();
+    const projectNamespace = configUtils.getProjectNamespace(projectXml);
 
-  return packageName;
-}
-
-// read visual studio project file which is actually a XML doc
-function readProject(projectPath) {
-  return new (xmldoc.XmlDocument)(fs.readFileSync(projectPath, 'utf8'));
-}
+    const directDependency = true;
 
-function findCppProject(folder, projectName) {
-  const cppProj = glob.sync(path.join('**', projectName + '.vcxproj'), {
-    cwd: folder,
-    ignore: ['node_modules/**', '**/Debug/**', '**/Release/**', '**/Generated Files/**', '**/packages/**'],
-  })[0];
+    const cppNamespace = projectNamespace.replace('.', '::');
+    const csNamespace = projectNamespace.replace('::', '.');
```

Comment:
careful with this, I was bitten by this last week - make sure you replace all instances since the default is to do just the first. Instead, do `.replace(/\./g, '::')` and `.replace(/::/g, '.')`